lhotse/cut.py

```diff
@@ -767,12 +801,19 @@ def load_features(self, mixed: bool = True) -> Optional[np.ndarray]:
             )
         if mixed:
             feats = mixer.mixed_feats
-            if self.num_frames - feats.shape[0] == 1:
-                # Edge case: both offset and cut duration in a given track ended with .5 (e.g. 10.5 and 1.5)
-                # and both of them were rounded down, resulting in an off-by-one error.
-                # We'll fix the padding here by repeating the last frame.
-                # (it's not elegant but I don't have another idea right now...)
-                feats = np.vstack([feats, feats[-1, :]])
+            # Note: The slicing below is a work-around for an edge-case
+            #  when two cuts have durations that ended with 0.005 (e.g. 10.125 and 5.715)
+            #  - then, the feature extractor "squeezed in" a last extra frame and the simple
+            #  relationship between num_frames and duration we strived for is not true;
+            #  i.e. the duration is 10.125 + 5.715 = 15.84, but the number of frames is
+            #  1013 + 572 = 1585. If the frame_shift is 0.01, we have gained an extra 0.01s...
+            if feats.shape[0] - self.num_frames == 1:
+                feats = feats[:self.num_frames, :]
+            # TODO(pzelasko): This can sometimes happen in a MixedCut with >= 5 different Cuts,
+            #   with a regular Cut at the end, when the mix offsets are floats with a lot of decimals.
+            #   For now we're duplicating the last frame to match the declared "num_frames" of this cut.
+            if feats.shape[0] - self.num_frames == -1:
+                feats = np.concatenate((feats, feats[-1:, :]), axis=0)
```

Comment:
Perhaps assert, after all of this, that feats.shape[0] == self.num_frames, so examples that differ by more than one crash early.